src/codegen/properties/attribute.py

```diff
@@ -0,0 +1,281 @@
+        """
+        `True`: This attribute can be reset back to default. This is also used to decide if a deleter has to be generated for the property.
+        `False`: This attribute cannot be reset back to default.
+        """
+        return self._resettable
+
+    @property
+    def type(self):
+        """
+        Data type of the attribute.
+        Here `enum` types are always represented as integers.
+        """
+        return self._type
+
+
+    @property
+    def has_explicit_write_buffer_size(self):
+        """
+        Specifies if an explicit write buffer size has to be provided when making the c function calls for the attribute.
+        If True then an additional uint parameter would be provided when calling the c function to mention the buffer size.
+        """
+        if not hasattr(self,"_has_explicit_write_buffer_size"):
+            return None
+        return self._has_explicit_write_buffer_size
+
+    @property
+    def object_has_factory(self):
+        """
+        This key is only present when `is_object` value is `True`.
+        'True' if the object has a factory implementation.
+        If the value is `True` then the `_factory` method is used for instantiation of the object.
+        """
+        return self._object_has_factory
+
+    @property
+    def is_enum(self):
+        """
+        Represents if the attribute is an enum or not.
+        """
+        return self._is_enum
+    
+    @property
+    def enum(self):
+        """
+        The enum type the attribute represents.
+        This key will only be available for an enum type attribute.
+        During code generation an attribute would be considered as an enum if it contains this key.
+        """
+        return self._enum
+
+    @property
+    def handle_parameters(self):
+        """
+        A list of parameters that represent handles that the attribute is part of.
+        These are used when defining the c function parameters, these are usually the first set of inputs to the function.
+        """
+        return self._handle_parameters
+
+    @property
+    def object_constructor_params(self):
+        """
+        This key is only present when `is_object` value is `True`.
+        The additional parameters that needs to included in the object creation apart from the attribute value.
+        During python code generation, these parameters are added as initial inputs when creating the object.
+        """
+        return self._object_constructor_params
+
+    @property
+    def python_class_name(self):
+        """
+        The name of the python class this attribute belongs to.
+        This is used to determine which attribute goes to which class when generating the python code.
+        """
+        return self._python_class_name
+
+    @property
+    def is_object(self):
+        """
+        'True': The attribute represents a known object.
+        'False' : The attribute represents the `type` value as usual.
```

Comment:
I don't know if doc tools like Sphinx will parse this correctly. https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html uses "type: description" for properties, not "value: description\nvalue2: description".

`ni-python-styleguide lint` probably warns about this.

Reply:
Updated the description as per the style guide.